proposals/js-promise-integration/Overview.md

```diff
@@ -2,299 +2,225 @@
 
-### Supporting Responsive Applications with Reentrancy
+Not all applications can equally tolerate being reentrant in this way. Certainly, languages in the C family do not make this straightforward. In fact, an application would typically have to have been engineered appropriately, by, for example, ensuring that each call to a suspending import does not interfere with globally shared state.
 
-A responsive application is able to respond to new requests even while suspended for existing ones. Note that we are not concerned with _multi-threaded_ applications (which can also be responsive): only one computation is expected to be active at any one time and all others would be _suspended_. Typically, such responsive applications are already crafted using an eventloop style architecture; even if they still use synchronous APIs.
+However, desktop applications, written for operating systems such as Mac OS and Windows, are often already structured in terms of an event loop that monitors input events and schedules UI effects. Such an application can often make good use of JSPI: perhaps by removing the application's event loop and replacing it with the browser's event loop.
 
-In fact, our example above is already technically re-entrant! However, it does suffer from a particular bug: if the export is reentered, while an existing call is suspended, the global variable holding the suspender may have the wrong value in it and that will result in a trap. We must ensure that the value of the `$suspender` global is properly maintained.
+## Specification
 
-This is necessary because the global variable used to communicate the identity of the suspender from the export call to the import is just that&mdash;global. It will be reset every time a call to the export is made. Only one task may be running at any one time, which means that the value of the `$suspender` global will not be changed while the task is running. However, the value of `$suspender` _will_ change if the task is suspended and a new task started. So, we need to ensure that the `$suspender` global is properly reset when a task is resumed.
+### WebIDL Interface
 
-The change involved is small, since the correct value of the `$suspender` global should be available to the resuming task. It was used in order to pass the suspender to the import, and, unless specifically dropped, is still available to the resuming task on the stack.
+```idl
+[Exposed=(Window,Worker,Worklet)]
+partial namespace WebAssembly {
 
-The required change is located in the call to `$compute_delta_import`:
+  Function promising(WebAssembly.Function fun);
```

Comment:
This looks like the only remaining dependency on the JS types proposal. To break that dependency, you'll need some other way to specify that this takes a Wasm function.

I don't think this needs to be handled in this PR, but I'd recommend opening an issue for it so it can be handled before this spec is finalized.